src/cudamatrix/cu-matrix.cc

```diff
@@ -2540,33 +2577,54 @@ void CuMatrixBase<Real>::Lookup(const std::vector<Int32Pair> &indices,
   // Checks the pointer.
   KALDI_ASSERT(output != NULL);
 
-  // Resizes the output vector.
-  output->resize(indices.size());
+#if HAVE_CUDA == 1
```

Comment:
This block inside the #if is wasteful in the case where CUDA is compiled in but not being used (e.g. no GPU available or not configured to use GPU), because it involves wasteful copying.  Inside the #if...#endif block you should have
if (CuDevice::Instantiate().Enabled()) {
} else
and then after the #endif, the next block can be inside braces {..}.  Make sure you understand why this works.